Update default signup form description & thank you copy

mmmavis opened the pull request:
Closes #4142 and #4143

Pretty straight-forward fixes. I think those weren't the troublesome tickets I thought they were 😂 

Note: @taisdesouzalessa and I worked together to tweak 'thank you' copy and design (removed the first paragraph to make the total copy shorter.

https://foundation-s-issue-4142-juuwq9.herokuapp.com/en/campaigns/late-possible-record-unit-season-organization/

mmmavis commented on source/js/components/join/join.jsx:
```diff
@@ -536,8 +557,7 @@ export default class JoinUs extends React.Component {
 JoinUs.defaultProps = {
   ctaHeader: `Protect the internet as a global public resource`,
-  ctaDescription: `Join our email list to take action and stay updated!`,
-  thankYouMessage: `If you haven’t previously confirmed a subscription to a Mozilla-related newsletter you may have to do so. <strong>Please check your inbox or your spam filter for an email from us.</strong>`,
```
Removed as none of the signup form has its own customized thank you message (e.g., none of the `.join-us` has `data-thank-...` attribute)

taisdesouzalessa commented:
@mmmavis the new Thank You message works great on:
- the footer
- the side form
in both desktop and mobile.

We have an issue for the one that appears after signing up for newsletter using the top nav button: the message is cut off. 
<img width="1235" alt="image" src="https://user-images.githubusercontent.com/22157921/74365927-e1a59080-4d83-11ea-97e6-f535e66f4025.png">

mmmavis commented:
PR is ready for re-review!

Pomax commented on source/js/components/join/join.jsx:
```diff
@@ -265,17 +267,34 @@ export default class JoinUs extends React.Component {
-            }}
-          />
+          <React.Fragment>
+            <p>
```
We'll need to wrap this in `getText` so that it can be localised.

Pomax commented on source/js/components/join/join.jsx:
```diff
@@ -265,17 +267,34 @@ export default class JoinUs extends React.Component {
+          <React.Fragment>
+            <p>
+              If you haven’t previously confirmed your opt-in to a
+              Mozilla-related email subscription you may have to do so now.{" "}
```
Does `&nbsp;` work rather than `{" "}`? If so, that'll make future localization a lot easier.

mmmavis replied:
Moved the strings to `locale-strings.jsx` and it seems like strings there are using `{" "}` already so I'm gonna keep the new strings that way.

mmmavis commented on source/js/components/petition/locale-strings.jsx:
```diff
@@ -122,7 +188,29 @@ export default {
     "Add my name": `Ajouter mon nom`,
-    "Sign up": `Je m’inscris`
+    "Sign up": `Je m’inscris`,
+    "If you haven’t previously confirmed your opt-in to a Mozilla-related email subscription you may have to do so now.": (
```
@Pomax do you know where we can get the actual localized strings from? Do we have a system in place that auto updates strings here?

Pomax replied:
No, these were all temporary work that we had localisers do for us because it was far less work to do the set of at-the-time fixed strings than it was to figure out a true localisation strategy. The work to do this properly is currently being investigated by @TheoChevalier

Pomax commented:
In principle but we should get those strings translated before we land this. I don't think it's wise to block this on the current work around adding proper localisation for React content, but we'll definitely need @TheoChevalier's help in making sure those few new strings have the correct localised versions set.

Pomax commented:
@TheoChevalier what would be required here to make sure we capture the string changes in this PR while we figure out how to do React content localization properly?

TheoChevalier commented:
> @TheoChevalier what would be required here to make sure we capture the string changes in this PR while we figure out how to do React content localization properly?

Not much from your side :) I’ve just exposed the strings to Pontoon, and I’m going to update this PR with the translations as they come in https://github.com/mozilla-l10n/engagement-l10n/commit/c718b5cb7be60c5f535a457b6c6cae1edb77633d

Let’s check back the translation coverage on Monday and decide if we’re good to go with that? Then I can open a follow-up PR if we get other translations later. Would that be okay?

TheoChevalier commented on source/js/components/join/join.jsx:
```diff
@@ -261,17 +263,21 @@ export default class JoinUs extends React.Component {
+          <React.Fragment>
+            <p>
+              {getText(`If you haven’t previously confirmed your opt-in to a
+              Mozilla-related email subscription you may have to do so now.`)}
```
This doesn’t work, because the leading spaces are considered part of the key here. So that string can’t be found.
Let’s probably just use a shorter key so that it fits in one line?

mmmavis replied:
Good point! Will update!

TheoChevalier commented:
Alright, we already have de, fr, pl, pt. We can merge this anytime now, with the key/identifier change I mentioned in my review.
If we ever get Spanish, I’ll make a follow up PR.

mmmavis commented:
@Pomax Please see my [last commit](https://github.com/mozilla/foundation.mozilla.org/pull/4192/commits/f3d3a6b64175d2deffaec1dbd953f7c2291a32a1). I've updated `CTAFactory` so that its `description` is more realistic (2 paragraphs instead of 6) and it's up to date with the DOM tag changes that I introduced in this PR. I've also updated `petition.jsx`  so it matches what we have for the signup component.

mmmavis commented:
This _only_ took 2 weeks to land 😂 😂 😂 

🎉  Merging now! Thanks, all!
